### PR TITLE
Add reset method

### PR DESCRIPTION
**Pull request by dmoranj**
Add a Reset method to the Outgoing message, so a COAP RST message can be returned to the caller.

### CODE REVIEW

**Comment by mcollina** [comment]
You have to add it to https://github.com/mcollina/node-coap/blob/master/lib/observe_write_stream.js too.

**Comment by dmoranj** [comment]
Added method to the Observe Write Stream.

**Comment by dmoranj** [comment]
This PR address issue #43

File: test/server.js [hunk 1]
```diff
@@ -215,6 +215,23 @@ describe('server', function() {
     })
   })
 
+  // TODO
```
Comment (mcollina) [inline test/server.js]: why is there a todo?

File: test/server.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ var coap      = require('../')
   , sinon     = require('sinon')
   , params    = require('../lib/parameters')
 
-describe('server', function() {
+describe.only('server', function() {
```
Comment (mcollina) [inline test/server.js]: why `only()`?

**Comment by mcollina** [comment]
Can you also add a test for the `ObserveWriteStream`? Thanks!

**Comment by mcollina** [comment]
Also, add yourself to https://github.com/mcollina/node-coap#contributors :).

File: test/server.js (hunk 2 above)
Comment (dmoranj) [inline test/server.js]: Because I forgot to remove it :D Same goes for the "TODO". God bless Code Reviews.

**Comment by dmoranj** [comment]
The test labeled "'should send a \'RST\' to the client if the msg.reset() method is invoked'" address the reset() function for the ObserveWriteSrteam.

**Comment by dmoranj** [comment]
Commit  3b6565c fixes the `only()` and "TODO"

**Comment by mcollina** [comment]
Released!

**Pull request merged**